homeassistant/helpers/service.py

```diff
@@ -165,16 +169,38 @@ def extract_entity_ids(hass, service_call, expand_group=True):
 async def _load_services_file(hass: HomeAssistantType, domain: str) -> JSON_TYPE:
     """Load services file for an integration."""
     integration = await async_get_integration(hass, domain)
+    services = {}
     try:
+        # For the python_script domain, look for 2 services.yaml files
+        if domain == PYTHON_SCRIPT_DOMAIN:
+            # 1. The integration's services.yaml file
+            services = await hass.async_add_executor_job(
+                load_yaml, str(integration.file_path / "services.yaml")
+            )
+
+            # 2. A user-provided services.yaml file
```

Comment:
I should probably move this into the component setup. I wouldn't need to modify this file at all in this pull request, and `python_script.reload` would also reload the user-provided service descriptions.

Reply:
Implemented in alternative pull request: https://github.com/home-assistant/home-assistant/pull/26069